lib/wallaroo/core/source/connector_source/connector_framed_source_notify.pony

```diff
@@ -183,13 +612,58 @@ class ConnectorSourceNotify[In: Any val]
       end
     end
 
-  fun ref accepted(source: ConnectorSource[In] ref) =>
+  //////////////////////////////
+  // Connection State Management
+  //////////////////////////////
+  fun ref accepted(source: ConnectorSource[In] ref, session_id: RoutingId) =>
     @printf[I32]((_source_name + ": accepted a connection\n").cstring())
+    if _fsm_state isnt _ProtoFsmDisconnected then
+      @printf[I32]("ERROR: %s.connected: state is %d\n".cstring(),
+        __loc.type_name().cstring(), _fsm_state())
+      Fail()
+    end
+    // reset/clear local state
+    _fsm_state = _ProtoFsmConnected
     _header = true
+    _session_active = true
+    _session_id = session_id
+    _clear_streams()
+    _credits = _max_credits
```

Comment:
This also happens when we receive a Hello msg.  Should this be happening in two places?

Reply:
Good question.  Keeping it here ought to be (at worst) harmless and (at best?) avoiding credit deficit for a misbehaving client?